### DISCUSSION TITLE
502 Bad Gateway, cannot acces on MeshCentral after about 5 minutes of connexion

### DISCUSSION THREAD

**Discussion by nyrvan1** [amidaware/tacticalrmm#1340] [General]
**Server Info (please complete the following information):**
 - OS: Debian 11
 - Browser: Brave
 - RMM Version (as shown in top left of web UI): 0.15.2

**Installation Method:**
  - [x ] Standard
  - [ ] Docker

**Agent Info (please complete the following information):**
- Agent version (as shown in the 'Summary' tab of the agent from web UI):
- Agent OS: Windows

**Describe the bug**
After the reboot of my server I can access to meshcentral but after about 5 minutes meshcentral deconnecte and when I refresh  the page I have an error web page: 502 Bad Gateway.  When I reboot the server again I can access to meshcentral but for about 5 minutes only. Here is  erro logs  on meshcentral :

-------- 28/10/2022 13:44:26 ---- 1.0.91 --------

/meshcentral/node_modules/mongodb/lib/sdam/topology.js:293
                const timeoutError = new error_1.MongoServerSelectionError(`Server selection timed out after ${serverSelectionTimeoutMS} ms`, this.description);
                                     ^

MongoServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at Timeout._onTimeout (/meshcentral/node_modules/mongodb/lib/sdam/topology.js:293:38)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7) {
  reason: TopologyDescription {
    type: 'Single',
    servers: Map(1) {
      '127.0.0.1:27017' => ServerDescription {
        address: '127.0.0.1:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1491001,
        lastWriteDate: 0,
        error: MongoNetworkError: connect ECONNREFUSED 127.0.0.1:27017
            at connectionFailureError (/meshcentral/node_modules/mongodb/lib/cmap/connect.js:387:20)
            at Socket.<anonymous> (/meshcentral/node_modules/mongodb/lib/cmap/connect.js:310:22)
            at Object.onceWrapper (node:events:628:26)
            at Socket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:157:8)
            at emitErrorCloseNT (node:internal/streams/destroy:122:3)
            at processTicksAndRejections (node:internal/process/task_queues:83:21) {
          [Symbol(errorLabels)]: Set(0) {}
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      }
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: null,
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined,
  [Symbol(errorLabels)]: Set(0) {}
}

-------- 28/10/2022 13:55:34 ---- 1.0.91 --------

/meshcentral/node_modules/mongodb/lib/cmap/connection.js:135
            op.cb(new error_1.MongoNetworkError(message));
                  ^

MongoNetworkError: connection 6 to 127.0.0.1:27017 closed
    at Connection.onClose (/meshcentral/node_modules/mongodb/lib/cmap/connection.js:135:19)
    at Socket.<anonymous> (/meshcentral/node_modules/mongodb/lib/cmap/connection.js:62:46)
    at Socket.emit (node:events:513:28)
    at TCP.<anonymous> (node:net:301:12) {
  [Symbol(errorLabels)]: Set(0) {}
}

So how to fix this trouble ?

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by wh1te909** [comment]
mongodb crashing, probably you don't have enough RAM

**Reply by nyrvan1** [reply]
On my server I have 20Gb of disque space and 1Gb of memory.  What is the ressource required ?

**Reply by nyrvan1** [reply]
And I only install TacticalRMM on this server.

**Reply by dinger1986** [reply]
https://docs.tacticalrmm.com/install_server/#hardware-os

3GB Ram

**Reply by nyrvan1** [reply]
Ok I see. Thank you for your time.
